components/edit-collective/MenuEditCollective.js

```diff
@@ -137,25 +147,38 @@ const MenuEditCollective = ({ collective, selectedSection }) => {
   const { formatMessage } = useIntl();
   const allSections = Object.values(EDIT_COLLECTIVE_SECTIONS);
   const displayedSections = allSections.filter(section => shouldDisplaySection(collective, section));
-  const displayedSectionsInfos = displayedSections.map(section => ({
+  const getSectionInfo = section => ({
     label: SECTION_LABELS[section] ? formatMessage(SECTION_LABELS[section]) : section,
     isSelected: section === selectedSection,
     section,
-  }));
+  });
+  const displayedSectionsInfos = displayedSections.map(getSectionInfo);
+
+  // eslint-disable-next-line react/prop-types
+  const renderMenuItem = ({ section, label, isSelected }) => (
+    <MenuItem
+      key={section}
+      selected={isSelected}
+      route="editCollective"
+      params={{ slug: collective.slug, section }}
+      data-cy={`menu-item-${section}`}
+    >
+      {label}
+    </MenuItem>
+  );
 
   return (
     <Flex width={0.2} flexDirection="column" mr={4} mb={3} flexWrap="wrap" css={{ flexGrow: 1, minWidth: 175 }}>
-      {sortBy(displayedSectionsInfos, 'label').map(({ section, label, isSelected }) => (
-        <MenuItem
-          key={section}
-          selected={isSelected}
-          route="editCollective"
-          params={{ slug: collective.slug, section }}
-          data-cy={`menu-item-${section}`}
-        >
-          {label}
-        </MenuItem>
-      ))}
+      {sortBy(displayedSectionsInfos, 'label').map(renderMenuItem)}
+      {collective.isHost && (
+        <React.Fragment>
+          <MenuDivider />
+          {collective.type === CollectiveType.COLLECTIVE &&
```

Comment:
`sectionsDisplayConditions` was setup to avoid this kind of checks that made the previous code hard to read. I'd be happy if we can avoid that. Maybe the right approach would be to categorize everything?

```
 General settings
... Info
... Payment methods
Features
... Conversations
... Collective goals
Host settings
... Invoices
```

Reply:
Although I agree it is more readable, I think simplicity should prevail over readability.
We can refactor the function that maps menu items, but ultimately I think that's way too much code to sort and display menu items.
Perhaps manual sorting and composable helper functions like `isHost()` `isCollective()` is enough?

Reply:
Manual sorting is indeed better (sorting was a side effect of https://github.com/opencollective/opencollective-frontend/pull/3321, not its main goal).

I agree that simplicity should prevail over readability. Consistency actually makes things more simple which is why I'm hesitant about mixing both approaches. However I understand that the code did not allow the separation of the menus and that you had to find a way to do that easily.

Let's figure out a way to improve this menu in another issue.

Reply:
Proposal for categories in https://github.com/opencollective/opencollective/issues/2832